SYCL/ESIMD/api/functional/value_conv.hpp

```diff
@@ -0,0 +1,117 @@
+//===-- value_conv.hpp - This file provides common functions generate values for
+//      testing. ----------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions that let obtain converted reference data for
+/// test according to current underlying types.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "value.hpp"
+
+namespace esimd_test::api::functional {
+
+// Utility class to retrieve specific values for tests depending on the source
+// and destination data types. May be used to retrieve converted reference data.
+// All provided methods are save to use and protected from UB when call
+// static_cast<int>(unsigned int).
+template <typename SrcT, typename DstT> struct value_conv {
+  static SrcT lowest() {
+    if constexpr (!sts::is_unsigned_v<SrcT> && sts::is_unsigned_v<DstT>) {
```

Comment:
```suggestion
    if constexpr (std::is_signed_v<SrcT> && std::is_unsigned_v<DstT>) {
```
A tiny misprint, here and elsewhere. Also why not to use `std::is_signed` instead of `!std::is_unsigned`?

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/696/commits/3ae1bec31097fff5e7647d14e3bac7e89b5f497a#diff-e031be0968a4e0cad81b5b2879e34920db09777fa3165f4f722fb4e0ab560826R28)